src/EnergyPlus/SetPointManager.cc

```diff
@@ -3703,6 +3780,253 @@ void GetSetPointManagerInputData(EnergyPlusData &state, bool &ErrorsFound)
+        if (setpointManager.CtrlTypeMode == SetPointManager::CtrlVarType::Invalid) {
+            // should not come here if idd type choice and key list is working
+            ShowSevereError(state, format("GetSetPointManagerInputs: {}=\"{}\", invalid field.", cCurrentModuleObject, cAlphaArgs(1)));
+            ShowContinueError(state, "..invalid " + cAlphaFieldNames(2) + "=\"" + cAlphaArgs(2) + "\".");
+            ShowContinueError(state, "..Valid values are \"Temperature\", \"MaximumTemperature\", or \"MinimumTemperature\".");
+            ErrorsFound = true;
+        }
+
+        setpointManager.SpAtLowRef = rNumericArgs(1);
+        setpointManager.SpAtHighRef = rNumericArgs(2);
+        setpointManager.LowRef = rNumericArgs(3);
+        setpointManager.HighRef = rNumericArgs(4);
+
+        setpointManager.RefNodeNum = GetOnlySingleNode(state,
+                                                       cAlphaArgs(3),
+                                                       ErrorsFound,
+                                                       DataLoopNode::ConnectionObjectType::SetpointManagerSystemNodeResetTemperature,
+                                                       cAlphaArgs(1),
+                                                       DataLoopNode::NodeFluidType::Blank,
+                                                       DataLoopNode::ConnectionType::Sensor,
+                                                       NodeInputManager::CompFluidStream::Primary,
+                                                       ObjectIsNotParent);
+
+        setpointManager.CtrlNodeListName = cAlphaArgs(4);
+        if (setpointManager.SpAtLowRef < setpointManager.SpAtHighRef) {
+            ShowWarningError(state, format("GetSetPointManagerInputs: {}=\"{}\", invalid setpoints.", cCurrentModuleObject, cAlphaArgs(1)));
+            ShowContinueError(state,
+                              format("...{}=[{:.1R}] is less than {}=[{:.1R}].",
+                                     cNumericFieldNames(1),
+                                     setpointManager.SpAtLowRef,
+                                     cNumericFieldNames(2),
+                                     setpointManager.SpAtHighRef));
+        }
+
+        NodeListError = false;
+
+        GetNodeNums(state,
+                    setpointManager.CtrlNodeListName,
+                    NumNodes,
+                    NodeNums,
+                    NodeListError,
+                    DataLoopNode::NodeFluidType::Blank,
+                    DataLoopNode::ConnectionObjectType::SetpointManagerSystemNodeResetTemperature,
+                    cAlphaArgs(1),
+                    DataLoopNode::ConnectionType::SetPoint,
+                    NodeInputManager::CompFluidStream::Primary,
+                    ObjectIsNotParent,
+                    _,
+                    cAlphaFieldNames(4));
+
+        if (!NodeListError) {
+            NumNodesCtrld = NumNodes;
```

Comment:
Not sure what the point is here to use `NumNodesCtrld` instead of `NumNodes` directly below. Or use `NumNodesCtrld` as the parameter in `GetNodeNums`. I know you're just following the pattern from elsewhere, so it's fine.